exercises/concept/take-a-number-deluxe/.docs/introduction.md

```diff
@@ -0,0 +1,143 @@
+# Introduction
+
+## GenServer
+
+`GenServer` (generic server) is a [behaviour][concept-behaviours] that abstracts common client-server interactions between Elixir processes.
+
+Remember the receive loop from when we learned about [processes][concept-processes]? The `GenServer` behaviour provides abstractions for implementing such loops, and for exchanging messages with a process that runs such a loop. It makes it easier to keep state and execute asynchronous code.
+
+Be warned that the name `GenServer` is loaded. It is also used to describe a _module_ that _uses_ the `GenServer` behaviour, as well as a _process_ that was started from a module that _uses_ the `GenServer` behaviour.
+
+The `GenServer` behaviour defines one required callback, `init/1`, and a few interesting optional callbacks: `handle_call/3`, `handle_cast/2`, and `handle_info/3`. The _clients_ using a `GenServer` aren't supposed to call those callbacks directly. Instead, the `GenServer` module provides functions that clients can use to communicate with a `GenServer` process.
+
+Often, a single module defines both a _client API_, a set of functions that other parts of your Elixir app can call to communicate with this `GenServer` process, and _server callback implementations_, which contain this `GenServer`'s logic.
+
+Let's take a look at a simple example of a `GenServer` first, and then learn what each callback means.
+
+### Examples
+
+This is an example server that can respond to the repetitive inquisitions of annoying passengers during a long road trip, more exactly the question: "are we there yet?". It keeps track of how many times this question has been asked, returning increasingly more annoyed responses.
+
+```elixir
+defmodule AnnoyingPassengerAutoreponder do
+  use GenServer
+  # Client API
+
+  def start_link(init_arg) do
+    GenServer.start_link(__MODULE__, init_arg)
+  end
+
+  def are_we_there_yet?(pid) do
+    GenServer.call(pid, :are_we_there_yet?)
+  end
+
+  # Server callbacks
+
+  @impl GenServer
+  def init(_init_arg) do
+    # the initial count of questions asked is always 0
+    state = 0
+    {:ok, state}
+  end
+
+  @impl GenServer
+  def handle_call(:are_we_there_yet?, _from, state) do
+    reply =
+      case state do
```

Comment:
`cond` might work better here? 
Not very important at all , feel free to ignore :)

Reply:
Yeah, it looks better thanks to no repetition of "n when n..."